lib/class-wp-theme-json-gutenberg.php

```diff
@@ -541,7 +541,7 @@ class WP_Theme_JSON_Gutenberg {
 	 * @var string[]
 	 */
 	const ELEMENTS = array(
-		'link'    => 'a:where(:not(.wp-element-button))', // The `where` is needed to lower the specificity.
+		'link'    => 'a:not(.wp-element-button)',
```

Comment:
There are a few other places the elements selectors might need updating as well as here if we are actually wanting to increase this selectors specificity from 0-0-1 to 0-1-1 e.g. the elements block support and the global styles output constant.

Given the spirit of this PR is to reduce specificity can we omit this change?

Reply:
I might be a bit turned around jumping between PRs, I take it this change was because at least in the theme.json class element selectors are [getting wrapped in `:where`](https://github.com/WordPress/gutenberg/pull/57841/files/a5c48bd0f87219bcce4dcdeadbdb0faff5099749#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R2576)?

As these selectors may be referenced elsewhere, is there a benefit to keeping the `a:where(:not(.wp-element-button))` selector for links so it is consistent with the elements block support?

Reply:
I'm not sure! Theoretically it would be nice if the element selectors didn't make any specificity assumptions, and we just manipulated the specificity as needed when using them in different contexts.

Relatedly, I think I found a bug in the link selector on this branch: the blocks-specific link selector no longer has the `:not` bit attached. Looking into it now!

Reply:
Oh turns out that's only recently been fixed on trunk, so rebasing should fix it. Might not be worth it if I'm going to split this into multiple PRs though.

Reply:
For posterity, the PR fixing it on trunk was https://github.com/WordPress/gutenberg/pull/59114.

Agreed, it's probably not work worrying about. On the splitting up front, I have a few other PRs based on this branch so my only question is do you see the different split parts as being pretty stable at this stage?

Reply:
I hope so, but this PR isn't getting a huge amount of feedback or testing so far, so part of my reason to split it up is hope that smaller PRs might be shippable faster. Stability is hard to predict without feedback! It works fine according to my testing, but I'm likely overlooking something 😅